# Gut microbiome diversity detected by high-coverage 16S & shotgun sequencing of paired stool/colon samples - Data Analysis

### By Joseph Luper Tsenum
Email: tsenumjosephluper@phystech.edu

Phystech School of Biological and Medical Physics (FBMF)

Moscow Institute of Physics and Technology (National Research University)

Russian Federation


### Introduction

The gut microbiome has a key role in human health and disease (Efimova et al. 2018). However, studying the complex structure and function of the gut microbiome through next generation sequencing is a challenging task and it results to reproducibility issues (Efimova et al. 2018)

### How the Efimova et al. 2018 sequenced the data

Here, Efimova et al. 2018 obtained cross-sectional colon biopsies and faecal samples from nine participants in their COLSCREEN study and sequenced them in high coverage through Illumina pair-end shotgun (for faecal samples) and IonTorrent 16S (for paired feces and colon biopsies) technologies (Efimova et al. 2018). The metagenomes comprised of between 47 and 92 million reads per sample and the targeted sequencing coverage was more than 300 k reads per sample across seven (7) hypervariable regions of the 16S gene (Efimova et al. 2018). Their data is publicly available together with the code for metagenomic analysis using up-to-date bioinformatics algorithms (Efimova et al. 2018). These results will contribute to the full understanding into developing 
all-inclusive microbiome analysis pipelines and also provide data for onward testing for complex gut microbiome analysis (Efimova et al. 2018).
